job/server/src/main/java/alluxio/job/plan/transform/format/JobPath.java

```diff
@@ -93,20 +96,28 @@ public JobPath(String scheme, String authority, String path) {
 
   @Override
   public FileSystem getFileSystem(Configuration conf) throws IOException {
-    try {
-      return CACHE.computeIfAbsent(new FileSystemKey(this, conf), (key) -> {
-        try {
-          return super.getFileSystem(conf);
-        } catch (IOException e) {
-          throw new RuntimeException(e);
-        }
-      });
-    } catch (RuntimeException e) {
-      Throwable cause = e.getCause();
-      if (cause instanceof IOException) {
-        throw (IOException) cause;
+    // Avoid caching at lower levels both because the key at this level has more properties and
+    // this cache wants to close the file systems that don't make it into this cache.
+    conf.setBoolean(ALLUXIO_HADOOP_FILESYSTEM_DISABLE_CACHE, true);
+
+    FileSystemKey key = new FileSystemKey(this, conf);
+    FileSystem fileSystem = CACHE.get(key);
+    if (fileSystem != null) {
+      return fileSystem;
+    }
+
+    fileSystem = super.getFileSystem(conf);
+
+    synchronized (CACHE) {
```

Comment:
I don't think you will need `synchronized`. Instead there should be a `putIfAbsent` which may be applicable, or a `compute` method.